js/main.js

```diff
@@ -245,101 +245,572 @@
-                    "incompatiblePermissions": [OC.PERMISSION_UPDATE],
-                    "incompatibleAttributes": [
-                        {
-                            "scope": OCA.Onlyoffice.AppName,
-                            "key": "review",
-                            "enabled": true
-                        },
-                        {
-                            "scope": OCA.Onlyoffice.AppName,
-                            "key": "fillForms",
-                            "enabled": true
-                        }
-                    ]
-                });
-            }
-
-            if (config.modifyFilter) {
-                model.registerShareAttribute({
-                    "scope": OCA.Onlyoffice.AppName,
-                    "key": "modifyFilter",
-                    "default": true,
-                    "label": t(OCA.Onlyoffice.AppName, "modify filter"),
-                    "requiredPermissions": [OC.PERMISSION_UPDATE],
-                    "incompatibleAttributes": [
-                        {
-                            "scope": OCA.Onlyoffice.AppName,
-                            "key": "commentOnly",
-                            "enabled": true
-                        }
-                    ]
-                });
-            }
-        }
-    };
+	OCA.Onlyoffice.ShareOptions = {
+
+		/**
+		 * @type {OCA.Share.ShareItemModel}
+		 */
+		model: null,
+
+		_shareOptions: [
+			{
+				name: "onlyoffice-download",
+				label: t(OCA.Onlyoffice.AppName, "download"),
+				div: 1,
+				attribute: {
+					scope: "permissions",
+					key: "download"
+				}
+			},
+			{
+				name: "onlyoffice-modify-filter",
+				label: t(OCA.Onlyoffice.AppName, "modify filter"),
+				div: 1,
+				attribute: {
+					scope: OCA.Onlyoffice.AppName,
+					key: "modifyFilter"
+				}
+			},
+			{
+				name: "onlyoffice-review",
+				label: t(OCA.Onlyoffice.AppName, "review"),
+				div: 2,
+				attribute: {
+					scope: OCA.Onlyoffice.AppName,
+					key: "review"
+				}
+			},
+			{
+				name: "onlyoffice-fill-forms",
+				label: t(OCA.Onlyoffice.AppName, "form filling"),
+				div: 2,
+				attribute: {
+					scope: OCA.Onlyoffice.AppName,
+					key: "fillForms"
+				}
+			},
+			{
+				name: "onlyoffice-comment",
+				label: t(OCA.Onlyoffice.AppName, "comment"),
+				div: 2,
+				attribute: {
+					scope: OCA.Onlyoffice.AppName,
+					key: "comment"
+				}
+			}
+		],
+
+		_shareOptionsTemplate: null,
+
+		/**
+		 * Extend ShareItemModel.addShare with onlyoffice attributes
+		 *
+		 * @param properties
+		 */
+		addShareProperties: function(properties) {
+			var extendedProperties = properties;
+
+			// get default permissions
+			extendedProperties.permissions = this.model.getDefaultPermissions();
+
+			// disable resharing as it is not supported currently
+			extendedProperties.permissions = this._removePermission(
+				extendedProperties.permissions, OC.PERMISSION_SHARE
+			);
```

Comment:
We need to make sure resharing is stable for share attributes (share merging scenarious, public links etc). This will be available in the future, lets keep it disabled for now.